pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -971,9 +972,31 @@ func (a *ApplicationBackupController) backupVolumes(backup *stork_api.Applicatio
 				return fmt.Errorf("error getting volume driver name: %v", err)
 			}
 			if driverName == volume.PortworxDriverName {
-				volumeInfos, err := driver.GetBackupStatus(backup)
+				batchCount := defaultGetBackupStatusBatchCount
+				volumes := backup.Status.Volumes
+				volumeInfos := make([]*stork_api.ApplicationBackupVolumeInfo, 0)
+				// Fetch the status of the volumes in batches
+				for i := 0; i < len(volumes); i += batchCount {
+					end := min(i+batchCount, len(volumes))
+					batchVolumes := volumes[i:end]
+					backup.Status.Volumes = batchVolumes
+					batchVolumeInfos, err := driver.GetBackupStatus(backup)
+					if err != nil {
+						logrus.Errorf("error getting backup status: %v", err)
+						return err
+					}
+					backup.Status.LastUpdateTimestamp = metav1.Now()
+					backup.Status.Volumes = volumes
```

Comment:
you could move this to a seperate func to reduce duplication:
```
batchVolumeInfos, err := a.processVolumeBatch(backup, driver, batchVolumes)
if err != nil {
log.ApplicationBackupLog(backup).Errorf("Failed to process volume batch for driver %v: %v", driverName, err)
return err
}
volumeInfos....

```
---
```
func (a *ApplicationBackupController) processVolumeBatch(
backup *stork_api.ApplicationBackup,
    driver volume.Driver,
    volumes []*stork_api.ApplicationBackupVolumeInfo,
) ([]*stork_api.ApplicationBackupVolumeInfo, error {
    backup.Status.Volumes = volumes
    batchVolumeInfos, err := driver.GetBackupStatus(backup)
    if err != nil {
        return nil, fmt.Errorf("failed to get volume status: %v", err)
    }

    backup.Status.LastUpdateTimestamp = metav1.Now()
    if err := a.client.Update(context.TODO(), backup); err != nil {
        return nil, fmt.Errorf("failed to update backup status: %v", err)
    }

    time.Sleep(batchProcessingDelayMs * time.Millisecond)
    return batchVolumeInfos, nil
    }

Reply:
Fixed , moved to a function